python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -668,22 +668,34 @@ def align(self, other):
         dtype: geometry)
 
         """
-        index = (
-            other.index if len(other.index) >= len(self.index) else self.index
-        )
+
+        # Merge the two indices and sort them
+        idx1 = cudf.DataFrame({"idx": self.index})
+        idx2 = cudf.DataFrame({"idx": other.index})
+        index = idx1.merge(idx2, how="outer").sort_values("idx")["idx"]
+        index.name = None
```

Comment:
I noticed when `len(other) == len(aligned_right) and len(self) == len(aligned_left)` this computation (which is expensive), is actually not needed. Can we rearrange the logic a bit to only compute `index` when needed?

Reply:
Possibly, I'll try it out.

Reply:
Great suggestion, thanks.

Reply:
Done.